### PR TITLE
fix(action-bar, combobox, dropdown, filter, input-date-picker, popover, tooltip): make methods using debounce/throttle private to prevent lodash type errors

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** NA

## Summary
@dasa reported a `lodash` type error from `calcite-tooltip` when bumping the JSAPI to `beta.94`. You can see lodash being imported in the types on line 59 of [`tooltip.d.ts`](https://unpkg.com/browse/@esri/calcite-components@1.0.0-beta.94/dist/types/components/tooltip/tooltip.d.ts).

Making the method private prevents the generated interface member from having a type defined. 

Edit:
Looks like there were additional components besides `tooltip`:
image

I added a bash command to Franco's type testing PR #5062 so this won't happen again 
```bash
! grep -rnw 'dist/types' -e '<reference types='
```

### CODE REVIEW

**Review by driskull — APPROVED** [review]

**Comment by driskull** [comment]
@benelan do we need to do this for other components? There are others that do the same. `action-bar`, `dropdown`, `filter`, etc.

**Comment by benelan** [comment]
Ah yes we do, I'll add those to this PR. We should try to make an ESLint rule somehow

File: src/components/combobox/combobox.tsx [hunk 1]
```diff
@@ -776,14 +776,14 @@ export class Combobox
...
-  emitCalciteLookupChange = debounce(this.internalCalciteLookupChangeEvent, 0);
+  private emitCalciteLookupChange = debounce(this.internalCalciteLookupChangeEvent, 0);
 
   internalComboboxChangeEvent = (): void => {
     const { selectedItems } = this;
     this.calciteComboboxChange.emit({ selectedItems });
   };
 
-  emitComboboxChange = debounce(this.internalComboboxChangeEvent, 0);
+  private emitComboboxChange = debounce(this.internalComboboxChangeEvent, 0);
```
Comment (benelan) [inline src/components/combobox/combobox.tsx:786]: @driskull before I merge, making the event emitters private won't break types for our users right?

Comment (jcfranco) [reply src/components/combobox/combobox.tsx:786]: This should be safe. The function invoking the emitter is being made private, not the emitter itself.

**Pull request merged**